src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -436,6 +459,163 @@ private protected void CompleteGesture()
 		}
 		#endregion
 
+		#region Drag And Drop (recognizer settings / custom bubbling / drag starting event)
+		private void UpdateDragAndDrop(bool isEnabled)
+		{
+			// Note: The drag and drop recognizer setting is only driven by the CanDrag,
+			//		 no matter which events are subscribed nor the AllowDrop.
+
+			var settings = _gestures.Value.GestureSettings;
+			settings &= ~GestureSettingsHelper.DragAndDrop; // Remove all configured drag and drop flags
+			if (isEnabled)
+			{
+				settings |= GestureSettings.Drag;
+			}
+
+			_gestures.Value.GestureSettings = settings;
+		}
+
+		partial void PrepareManagedDragAndDropEventBubbling(RoutedEvent routedEvent, ref RoutedEventArgs args, ref BubblingMode bubblingMode)
+		{
+			switch (routedEvent.Flag)
+			{
+				case RoutedEventFlag.DragStarting:
+				case RoutedEventFlag.DropCompleted:
+					// Those are actually not routed events :O
+					bubblingMode = BubblingMode.StopBubbling;
+					break;
+
+				case RoutedEventFlag.DragEnter when !CanDrag:
+				case RoutedEventFlag.DragOver when !CanDrag:
+				case RoutedEventFlag.DragLeave when !CanDrag:
+				case RoutedEventFlag.Drop when !AllowDrop:
+					// The Drag and Drop "routed" events are raised only on controls that opted-in
+					bubblingMode = BubblingMode.IgnoreElement;
+					break;
+			}
+		}
+
+		private void OnDragging(DraggingEventArgs args)
+		{
+			if (args.DraggingState != DraggingState.Started // This UIElement is actually interested only by the starting
+				|| !CanDrag // Sanity ... should never happen!
+				|| !args.Pointer.Properties.IsLeftButtonPressed)
+			{
+				//if (CoreWindow.GetForCurrentThread()!.LastPointerEvent is PointerRoutedEventArgs ptArgs
+				//	&& ptArgs.Pointer.PointerDeviceType == args.PointerDeviceType)
+				//{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6270104)